core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -253,6 +286,27 @@ public void run() {
       }
     };
 
+  /** Returns value of security property of network address cache ttl. */
+  private long getNetworkAddressCacheTtl() {
+    String cacheTtlPropertyValue = Security.getProperty(NETWORKADDRESS_CACHE_TTL_SECURITY_PROPERTY);
```

Comment:
I'm a little uncomfortable parsing this each time.  Why not parse it once at class init?